Userland/Libraries/LibVideo/Containers/Matroska/MatroskaDemuxer.h

```diff
@@ -18,6 +18,8 @@ class MatroskaDemuxer final : public Demuxer {
     // FIXME: We should instead accept some abstract data streaming type so that the demuxer
     //        can work with non-contiguous data.
     static DecoderErrorOr<NonnullOwnPtr<MatroskaDemuxer>> from_file(StringView filename);
+    static DecoderErrorOr<NonnullOwnPtr<MatroskaDemuxer>> from_mapped_file(NonnullRefPtr<Core::MappedFile> const& mapped_file);
```

Comment:
`NonnullRefPtr<Core::MappedFile> const&` as a parameter is an awkward anti-pattern. Especially here, where we actually want to give another (new) object a reference to the MappedFile.

This function should take the mapped_file by value and move() it into the `Reader::from_mapped_file` call.